source/simulator/assemblers/newton_stokes.cc

```diff
@@ -166,7 +173,13 @@ namespace aspect
                                                  // approximation in the bottom right needs to
                                                  // only be scaled by 1/eta, without considering
                                                  // the derivatives
-                                                 one_over_eta
+                                                 ( one_over_eta +
+                                                   // add the dilation LHS term if plastic dilation
+                                                   // is enabled
+                                                   ( prescribed_dilation != nullptr ?
+                                                     prescribed_dilation->dilation_lhs_term[q] :
+                                                     0.0 )
+                                                 )
```

Comment:
please do this computation up in line 146 where we compute `one_over_eta`. E.g.:
`const double one_over_eta_plus_dilation = ...`

The current loop is one of the most often executed lines in ASPECT, so it is better to move computations and if-branches outside of it.